dandiapi/api/tasks/__init__.py

```diff
@@ -12,10 +13,18 @@
     write_dandiset_yaml,
 )
 from dandiapi.api.models import Asset, AssetBlob, Version
+from dandiapi.api.models.dandiset import Dandiset
 
 logger = get_task_logger(__name__)
 
 
+class UnembargoTask(Task):
+    def on_failure(self, *task_args, **task_kwargs):
```

Comment:
I would probably just do this logic inline rather than as a base task, since I don't see this class being reused for other tasks.

Reply:
Good point, not sure why I didn't think of that. This makes everything much simpler and easy to test!